cmd/ingress-operator/start.go

```diff
@@ -38,26 +38,35 @@ const (
 	// operator's namespace that will hold the credentials that the operator
 	// will use to authenticate with the cloud API.
 	cloudCredentialsSecretName = "cloud-credentials"
-	// defaultTrustedCABundle is the fully qualified path of the trusted CA bundle
-	// that is mounted from configmap openshift-ingress-operator/trusted-ca.
-	defaultTrustedCABundle = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+
+	// defaultShutdownFile is the fully qualified path of the trusted CA bundle
+	//that is mounted from configmap openshift-ingress-operator/trusted-ca.
+	defaultShutdownFile = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
 )
 
 func NewStartCommand() *cobra.Command {
-	return &cobra.Command{
+	var options struct {
+		ShutdownFile string
+	}
+
+	cmd := &cobra.Command{
 		Use:   "start",
 		Short: "Start the operator",
 		Long:  `starts launches the operator in the foreground.`,
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := start(); err != nil {
+			if err := start(options.ShutdownFile); err != nil {
 				log.Error(err, "error starting")
 				os.Exit(1)
 			}
 		},
 	}
+
+	cmd.Flags().StringVarP(&options.ShutdownFile, "shutdown-file", "s", defaultShutdownFile, "file which when modified should trigger a shutdown")
```

Comment:
What do you think about making this a slice?
```go
cmd.Flags().StringSliceVarP(&options.ShutdownFiles, "shutdown-file", "s", []string{defaultTrustedCABundle}, "file which when modified should trigger a shutdown")
```
For your use-case, instead of specifying an empty file name, you would specify an empty list of files to watch, which seems semantically cleaner to me, and if for whatever reason you needed to specify multiple files, you could.

Reply:
I started with that, but then the logic gets even bigger, and we don't have any use case for multiple, so it didn't seem worth it

Reply:
Although that was before I switched to always reading from the watch channel and using the option to control whether to add stuff to the watch, so using a list would probably be only nominally more logic. Still not sure it matters though